feat: add debug rpc method to dump paymaster balances from pool

dancoombs opened the pull request:
Closes #573 

## Proposed Changes

  - Add new debug RPC method to dump and paymaster balances
  - Wire it all up

dancoombs commented:
Likely going to remove the "check" endpoint from this, will just do it offline.

alex-miao reviewed (approved):
Looks good, only small thing is that we should add this method in `docs/architecture/rpc.md`

dancoombs commented:
> Looks good, only small thing is that we should add this method in `docs/architecture/rpc.md`

Good call, added some docs and cleaned a few things up

dancoombs commented:
Output looks like this:

```
curl 'localhost:3000' --header 'Content-Type: application/json' --data '{"method":"debug_bundler_dumpPaymasterBalances","params":["0x5FF137D4b0FDCD49DcA30c7CF57E578a026d2789"],"id":42,"jsonrpc":"2.0"}'
{"jsonrpc":"2.0","result":[{"address":"0xfb10850286e8f6b7b66bd60fb50b643afb8f13b1","pendingBalance":"0x316f486b5da791287","confirmedBalance":"0x316f486b5da791287"}],"id":42}% 
```

0xfourzerofour commented on crates/pool/src/server/local.rs:
```diff
@@ -550,27 +608,15 @@ where
-                                },
+                        ServerRequestKind::GetReputationStatus{ entry_point, address } => {
+                            match self.get_reputation_status(entry_point, address) {
+                                Ok(status) => Ok(ServerResponse::GetReputationStatus {  status }),
```
nit: fix spacing in `status`

0xfourzerofour commented on crates/pool/src/server/local.rs:
```diff
@@ -447,6 +469,28 @@ where
+        response: oneshot::Sender<Result<ServerResponse, PoolServerError>>,
+        f: F,
+    ) where
+        F: FnOnce(Arc<M>, oneshot::Sender<Result<ServerResponse, PoolServerError>>) -> Fut,
```
this cool

0xfourzerofour reviewed (approved):
Good work on this
